boilerplate/openshift/golang-osd-operator/standard.mk

```diff
@@ -153,11 +161,34 @@ go-generate:
 	${GOENV} go generate $(TESTTARGETS)
 	# Don't forget to commit generated files
 
+# go-get-tool will 'go install' any package $2 and install it to $1.
+define go-get-tool
+@{ \
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(shell dirname $(1)) go install $(2) ;\
+echo "Installed in $(1)" ;\
+rm -rf $$TMP_DIR ;\
+}
+endef
+
+CONTROLLER_GEN = $(PWD)/bin/controller-gen
+CONTROLLER_GEN_VERSION = v0.8.0
+ifeq ($(USE_OLD_SDK), TRUE)
+CONTROLLER_GEN_VERSION = v0.3.0
+endif
+
+.PHONY: controller-gen
+controller-gen: ## Download controller-gen locally if necessary.
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION))
+
 .PHONY: op-generate
-op-generate:
-	# The artist formerly known as `operator-sdk generate crds`:
+op-generate: controller-gen
 ifeq ($(CRD_VERSION), v1beta1)
-	cd pkg/apis; controller-gen crd paths=./... output:dir=../../deploy/crds
+	cd $(API_DIR); $(CONTROLLER_GEN) crd paths="./..." output:dir=$(PWD)/deploy/crds
```

Comment:
It looks like L191 is the same as L205, except for some quotes. Do those really matter? If not, suggest putting it outsid of the conditional and deduplicating.

Reply:
Done